supervisor/shared/background_callback.c

```diff
@@ -33,7 +33,7 @@
 #include "supervisor/shared/tick.h"
 #include "shared-bindings/microcontroller/__init__.h"
 
-STATIC volatile background_callback_t *callback_head, *callback_tail;
+STATIC volatile background_callback_t * volatile callback_head, * volatile callback_tail;
```

Comment:
the pointer should be volatile rather than the data it is pointed to. We actually can have it as

```
STATIC background_callback_t * volatile callback_head, * volatile callback_tail;
```

But I will leave it as both volatile just in case I misunderstand its purpose.